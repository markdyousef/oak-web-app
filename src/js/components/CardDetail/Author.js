// @flow
import React from 'react';
import styled from 'styled-components';
import Avatar from '../shared/Avatar';
import type { Creator } from './types';

const Container = styled.div`
    height: 54px;
    display: flex;
    align-items: center;
    margin-bottom: 32px;
`;

const Name = styled.h3`
    max-width: 500px;
    font-size: 20px;
    margin-left: 16px;
`;

type Props = {
    creator?: Creator
};

export default({ creator }: Props) => {
    let picture;
    // prefer avatar over gravatar
    if (creator && creator.gravatar) picture = creator.gravatar;
    if (creator && creator.avatar) picture = creator.avatar.urlThumb64;
    return (
        <Container>
            <Avatar img={picture} alt="author" size="54px" />
            <Name>{creator && creator.username}</Name>
        </Container>
    );
};
