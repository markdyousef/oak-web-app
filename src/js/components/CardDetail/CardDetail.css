.modal {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    z-index: 2000;
    background-color: rgba(0, 0, 0, 0.4);
    display: flex;
    align-items: center;
    justify-content: center;
}

.container {
    width: 600px;
    height: 600px;
    background-color: #fff;
    border-radius: 3px;
    padding: 20px;
    border: 1px solid #ddd;
}
