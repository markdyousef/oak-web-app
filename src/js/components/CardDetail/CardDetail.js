// @flow
import React, { Component, PropTypes } from 'react';
import Description from '../shared/Description';
import TopBar from './TopBar';

import css from './CardDetail.css';

class CardDetail extends Component {
    static propTypes = {
        router: PropTypes.shape({
            goBack: PropTypes.func.isRequired
        }).isRequired
    }
    constructor() {
        super();
        this.onSave = this.onSave.bind(this);
        this.state = {
            showEdit: false
        };
    }
    onSave() {
        this.setState({ showEdit: false });
    }
    render() {
        const { showEdit } = this.state;
        const { router } = this.props;
        return (
            <div className={css.container}>
                <TopBar
                    close={() => router.goBack()}
                    save={this.onSave}
                    showEdit={showEdit}
                    edit={() => this.setState({ showEdit: true })}
                />
                <div className={css.main}>
                    {/* <div className={css.description}>
                        <Description />
                    </div> */}
                </div>
            </div>
        );
    }
}

export default CardDetail;
