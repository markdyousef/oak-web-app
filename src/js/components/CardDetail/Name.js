// @flow
import React from 'react';
import styled from 'styled-components';

const Input = styled.input`
    font-size: 24px;
    border: 0;
    padding-bottom: 10px;
`;

export default ({ onChange, name, placeholder, readOnly }) => (
    <Input
        onChange={event => onChange(event.target.value)}
        value={name}
        placeholder="Titel"
        readOnly={readOnly}
    />
);
