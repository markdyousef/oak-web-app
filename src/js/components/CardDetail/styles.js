// @flow
import styled from 'styled-components';
import colors from '../../styles/colors';

export const Container = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${colors.white};
`;

export const ContainerInner = styled.div`
    position: relative;
    ${''/* display: flex;
    flex-direction: column;
    flex-grow: 1;
    min-width: 1px; */}
`;

export const Main = styled.section`
    margin-top: 60px;
    overflow: hidden;
    display: flex;
    flex: 1;
`;

export const EditorContainer = styled.div`
    position: relative;
    margin: 0 auto;
    max-width: 1000px;
    width: 100%;
    display: flex;
    flex: 1;
    height: 100%;
    overflow: scroll;
    height: 100vh;
    padding: 50px 0 30px 0;
    @media (max-width: 1200px) {
        padding: 50px 40px 30px 160px;
    }
    &::-webkit-scrollbar {
        display: none;
    }
`;
