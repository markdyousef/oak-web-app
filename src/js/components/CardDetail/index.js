import CardDetail from './CardDetail';
import wrapper from './CardDetailWrapper';

export default CardDetail;

export {
    CardDetail,
    wrapper
};
