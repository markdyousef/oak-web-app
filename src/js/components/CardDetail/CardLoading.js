import React from 'react';
import { Container, ContainerInner, Main, EditorContainer } from './styles';

export default () => {
    return (
        <Container>
            <ContainerInner>
                <Main>
                    <EditorContainer>
                        <h1>LOADING</h1>
                    </EditorContainer>
                </Main>
            </ContainerInner>
        </Container>
    );
};
