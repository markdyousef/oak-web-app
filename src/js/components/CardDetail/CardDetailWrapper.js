// @flow
import React, { Component } from 'react';
import { EditorState } from 'draft-js';
import { is } from 'immutable';
import { uploadImage } from '../../utils';
import type { DefaultProps, State, Props } from './types';
import CardLoading from './CardLoading';

export default (CardDetail:Function) => {
    return (
        class Wrapper extends Component<DefaultProps, Props, State> {
            static defaultProps: DefaultProps;
            state: State;
            props: Props;
            componentWillMount() {
                const { params, updateCard, updateComments } = this.props;
                if (params.collectionId) {
                    updateCard({
                        key: 'collectionId',
                        value: params.collectionId
                    });
                    if (params.cardId) {
                        updateCard({
                            key: 'cardId',
                            value: params.cardId
                        });
                    }
                }
                if (params.comments) {
                    updateComments({
                        key: 'showComments',
                        value: true
                    });
                }
            }
            componentWillReceiveProps(nextProps: Props) {
                const { loading } = nextProps;
                if (loading) {
                    this.updateCard('isLoading', true);
                } else {
                    this.updateCard('isLoading', false);
                }
            }
            componentWillUnmount() {
                const { clearCard } = this.props;
                clearCard();
            }
            updateCard = (key: string, value: any) => this.props.updateCard({ key, value })
            onShowLabels = (show?:bool) =>
                this.props.updateLabels({
                    key: 'showLabels',
                    value: show
                })
            addFile = (file: Object, type?: string) => {
                const { addImage } = this.props;
                return new Promise((resolve, reject) => {
                    if (!type || type === 'image') {
                        uploadImage(file)
                            .then((res) => {
                                const image = { ...res, type: 'image', file };
                                addImage(image);
                                resolve(image);
                            })
                            .catch(() => {
                                const message = {
                                    type: 'error',
                                    message: "File couldn't be added",
                                    onClick: () => this.addFile(file)
                                };
                                this.setState({ message });
                                reject(message);
                            });
                    }
                });
            }
            onChange = (editorState:EditorState) => {
                const { card } = this.props;
                // check if card has been edited
                const content = card.get('editorState').getCurrentContent();
                const isEqual = is(editorState.getCurrentContent(), content);
                if (!isEqual) this.updateCard('isEdited', true);

                this.updateCard('editorState', editorState);
            }
            render() {
                const { card, comments, showLabels, isLoading, seed } = this.props;

                // loading state
                if (isLoading) return <CardLoading />;


                return (
                    <CardDetail
                        onChange={this.onChange}
                        addFile={this.addFile}
                        onCloseError={() => this.updateCard('message', { type: 'error', message: 'close' })}
                        existingCard={card.get('cardId')}
                        showComments={comments.get('showComments')}
                        collectionId={card.get('collectionId')}
                        showLabels={showLabels}
                        onShowLabels={this.onShowLabels}
                        changeName={name => this.updateCard('name', name)}
                        editorState={card.get('editorState')}
                        readOnly={card.get('readOnly')}
                        creator={seed && seed.creator}
                    />
                );
            }
        }
    );
};
