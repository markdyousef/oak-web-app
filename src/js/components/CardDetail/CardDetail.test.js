import React from 'react';
import { shallow } from 'enzyme';
import CardDetail from './CardDetail';

it('renders correctly', () => {
    const props = {
        close: () => {}
    };
    const shallowComponent = shallow(
        <CardDetail {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
