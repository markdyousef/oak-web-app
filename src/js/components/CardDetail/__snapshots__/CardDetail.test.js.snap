exports[`test renders correctly 1`] = `
<div
  className="kFXINv">
  <nav
    className="iBrQER">
    <div
      className="epePFJ">
      <div
        className="iefG">
        <div>
          <button
            className="bBPMRg"
            onClick={[Function]}
            type={null}>
            Add label
          </button>
        </div>
        <button
          className="iZsHgX"
          onClick={[Function]}
          type="primary">
          Save card
        </button>
      </div>
      <button
        className="iCZyJl"
        onClick={[Function]}
        type={null}>
        Comments
      </button>
      <button
        className="iCZyJl"
        onClick={[Function]}
        type={null}>
        X
      </button>
    </div>
  </nav>
  <section
    className="eqsUgR">
    <div
      className="cFuTfS">
      <div
        className="bycdYO"
        onClick={[Function]}>
        <div
          className="hSmQAk">
          <div
            className="DraftEditor-root">
            <div
              className="public-DraftEditorPlaceholder-root">
              <div
                className="public-DraftEditorPlaceholder-inner"
                id="placeholder-4j80e">
                write something
              </div>
            </div>
            <div
              className="DraftEditor-editorContainer">
              <div
                aria-activedescendant={undefined}
                aria-autocomplete={undefined}
                aria-describedby="placeholder-4j80e"
                aria-expanded={undefined}
                aria-haspopup={undefined}
                aria-label={undefined}
                aria-owns={undefined}
                className="public-DraftEditor-content"
                contentEditable={true}
                data-testid={undefined}
                onBeforeInput={[Function]}
                onBlur={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onCopy={[Function]}
                onCut={[Function]}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                onPaste={[Function]}
                onSelect={[Function]}
                role="textbox"
                spellCheck={true}
                style={
                  Object {
                    "outline": "none",
                    "whiteSpace": "pre-wrap",
                    "wordWrap": "break-word",
                  }
                }
                suppressContentEditableWarning={true}
                tabIndex={undefined}>
                <div
                  data-contents="true">
                  <div
                    className=""
                    data-block={true}
                    data-editor="4j80e"
                    data-offset-key="dene-0-0">
                    <div
                      className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="dene-0-0">
                      <span
                        data-offset-key="dene-0-0"
                        style={Object {}}>
                        <br
                          data-text="true" />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <input
            accept="image/*"
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file" />
        </div>
      </div>
    </div>
  </section>
</div>
`;
